Source/WebCore/css/StyleProperties.cpp

```diff
@@ -1621,6 +1621,19 @@ StringBuilder StyleProperties::asTextInternal() const
             shorthands.append(substitutionValue.shorthandPropertyId());
             value = substitutionValue.shorthandValue().cssText();
         } else {
+            // FIXME: could probably use matchingShorthandsForLonghand() instead of populating 'shorthands' manually.
+            switch (propertyID) {
+            case CSSPropertyCustom:
+            case CSSPropertyDirection:
+            case CSSPropertyUnicodeBidi:
+                // These are the only longhands not included in the 'all' shorthand.
+                break;
```

Comment:
Given that we had to write out this list by hand, can we write a test that will fail if this list is incorrect? If I remove something from or add something to this list, the test should fail.

Reply:
This is covered by the test that I added. I guess the only case that is not tested is if the CSSWG creates another longhand which is not not included in `all`, then it should be added to this list but it won't be detected.

Reply:
How would the test you added fail if I (incorrectly) added `CSSPropertyBorderRadius` to this switch statement?

Reply:
Well, that's a shorthand, and we shouldn't be getting shorthands here (even if we do in broken cases). But if for example I add the longhand `CSSPropertyBorderTopLeftRadius` to the switch, then:
```
FAIL 'all' shorthand alone assert_equals: expected "all: inherit;" but got "all: inherit; border-radius: inherit;"
FAIL 'all' shorthand with 'width' and 'height' assert_equals: expected "all: inherit;" but got "all: inherit; border-radius: inherit;"
FAIL 'all' shorthand with 'direction' and 'unicode-bidi' assert_equals: expected "direction: ltr; all: inherit; unicode-bidi: plaintext;" but got "direction: ltr; all: inherit; border-radius: inherit; unicode-bidi: plaintext;"
FAIL 'all' shorthand with 'width', 'height' and custom properties assert_equals: expected "--a: a; all: inherit; --b: b;" but got "--a: a; all: inherit; border-radius: inherit; --b: b;"
FAIL 'all' shorthand with all longhands assert_equals: expected "all: inherit; direction: inherit; unicode-bidi: inherit;" but got "all: inherit; border-radius: inherit; direction: inherit; unicode-bidi: inherit;"
```

Reply:
I added one test checking that the only standard longhands not included in `all` are `direction` and `unicode-bidi`. So the CSSWG adds another longhand like that, the test will need to be updated, and then it will fail if Webkit implements the new longhand but forgets to add it into this switch. The test also catches a bug in Blink with `-webkit-border-image`, so it's good.

Reply:
That new test sounds good. Is it already in this pull request, or coming in a future commit?

Reply:
Yes, it's already in the pull request:
```js
test(function() {
  let cssText = "all: inherit; ";
  for (let longhand of getComputedStyle(document.documentElement)) {
    cssText += longhand + ": inherit; ";
  }
  style.cssText = cssText;
  assert_equals(style.cssText, "all: inherit; direction: inherit; unicode-bidi: inherit;");
}, "'all' shorthand with all longhands");
```